src/dataframe/dataframe.jl

```diff
@@ -47,7 +47,10 @@ DataFrame(::GroupedDataFrame)
   to `false` then the constructor will still copy the passed columns
   if it is not possible to construct a `DataFrame` without materializing new columns.
 
-All columns in `columns` should have the same length.
+All columns in `columns` should be `AbstractVector`s and have the same length.
+An exception are `DataFrame(kwargs...)` and `DataFrame(pairs::Pair{Symbol,<:Any}...)`
+form constructors which additionally allow a column to be of any other type that is not
+an `AbstractArray`, in which case the passed value is implicitly broadcasted.
```

Comment:
Maybe just say "repeated"? Broadcast isn't actually used, even for structure which may support it.
```suggestion
an `AbstractArray`, in which case the passed value is automatically repeated to fill
a new vector of the appropriate length.
```